hack/tools/check-go-version.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+GO_MAJOR=${GO_MAJOR:-1}
+GO_MINOR=${GO_MINOR:-22}
+GO_PATCH=${GO_PATCH:-5}
```

Comment:
Could this be derived from the root `go.mod`? Or do we need a hardcoded version that is separately bumped/maintained?

Reply:
We need to check the version in the root go.mod... so we need a separate input for this. The values come from the line in the `Makefile`, so you don't have to bump the version in this file. We could make it a variable in the Makefile near the top.